rules/CodingStyle/Rector/Enum_/EnumCaseToPascalCaseRector.php

```diff
@@ -122,43 +121,54 @@ private function refactorClassConstFetch(ClassConstFetch $classConstFetch): ?Nod
             return null;
         }
 
-        if ($this->nodeTypeResolver->getType($classConstFetch->class)->isEnum()->no()) {
+        $constName = $classConstFetch->name->toString();
+        $pascalCaseName = $this->convertToPascalCase($constName);
+
+        // short circuit if already in pascal case
+        if ($constName === $pascalCaseName) {
             return null;
         }
 
-        $constName = $classConstFetch->name->toString();
+        $classReflection = $this->nodeTypeResolver->getType($classConstFetch->class)
+            ->getObjectClassReflections()[0] ?? null;
+
+        if ($classReflection === null || ! $classReflection->isEnum()) {
+            return null;
+        }
 
-        // Skip "class" constant
-        if ($constName === 'class') {
+        if (! $this->isEnumCase($classReflection, $constName, $pascalCaseName)) {
             return null;
         }
 
-        $enumClassName = $classConstFetch->class->toString();
-        if (! $this->reflectionProvider->hasClass($enumClassName)) {
+        if ($this->isUsedOutsideOfProject($classConstFetch->class)) {
```

Comment:
slef should can be recefenced to current classreflection name, if it is ThisType, should be utilze `$type->getStaticObjectType()` so can look up correct target name.

Reply:
Ok, let see it in the wild :), 

/cc @marcelthole @boesing in case you have a chance to test latest `"rector/rector": "dev-main"` for `EnumCaseToPascalCaseRector` rule :)